packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -1029,14 +1029,14 @@ export abstract class BaseService extends Resource
     return {
       attachToApplicationTargetGroup(targetGroup: elbv2.ApplicationTargetGroup): elbv2.LoadBalancerTargetProps {
         targetGroup.registerConnectable(self, self.taskDefinition._portRangeFromPortMapping(target.portMapping));
-        return self.attachToELBv2(targetGroup, target.containerName, target.portMapping.containerPort);
+        return self.attachToELBv2(targetGroup, target.containerName, target.portMapping.containerPort!);
```

Comment:
Sticking a `!` in here feels a bit dangerous to me. Can you add runtime checks just to be safe?

Also -- let's say the port mapping is a range. What happens then?

Reply:
>Sticking a ! in here feels a bit dangerous to me. Can you add runtime checks just to be safe?

See answer below for the reason why I didn't. If you still think it's worth adding the runtime checks to be future-proof, let me know

>Also -- let's say the port mapping is a range. What happens then?

It cannot happen because [`this.taskDefinition._validateTarget(options)`](https://github.com/aws/aws-cdk/blob/e4e08320412cd6cc999fb9eef0cd93b0445abceb/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L563) calls [`targetContainer.findPortMapping(targetContainerPort)`](https://github.com/aws/aws-cdk/blob/e4e08320412cd6cc999fb9eef0cd93b0445abceb/packages/aws-cdk-lib/aws-ecs/lib/container-definition.ts#L680) to look for a port mapping matching the specified `containerPort`. If no mapping is found, then `undefined` is returned and an error is thrown [here](https://github.com/aws/aws-cdk/blob/e4e08320412cd6cc999fb9eef0cd93b0445abceb/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L573).